### ISSUE TITLE
Future produced by IO is untyped

### ISSUE THREAD

**Issue by gcanti** [funfix/funfix.js#64]
Repro

```ts
// x: IO<number>
const x = IO.pure(1)
// y: Future<{}>, should be Future<number> I guess
const y = x.run()
```

I think there's a unnecessary type parameter `A` here https://github.com/funfix/funfix/blob/eacf892d1f1de14bc1b01c4bef04a89cf4b09ffb/packages/funfix-effect/src/io.ts#L390

**Comment by alexandru** [comment]
Thanks for noticing that @gcanti, not sure how I missed it.

**Comment by alexandru** [comment]
Released fix in `v6.2.1`
